### PR TITLE
Synchronous open-loop C++ test

### PR DESCRIPTION
**Pull request by vjpai**
This PR supplies a synchronous open-loop C++ test that supports various load distributions (Poisson, Pareto, etc) efficiently using a pre-calculated random-table with thread-specific iterators.

### CODE REVIEW

**Comment by vjpai** [comment]
It's been a while since I made this branch so there are merge conflicts galore; I'll deal with them in the afternoon.

**Comment by vjpai** [comment]
@nicolasnoble @ctiller I'm mentioning you here because my merge dragged both of you into this PR and got Googlebot upset. Not sure why.

File: test/cpp/qps/client_sync.cc [hunk 1]
```diff
@@ -64,17 +66,70 @@ namespace testing {
 
 class SynchronousClient : public Client {
  public:
-  SynchronousClient(const ClientConfig& config) : Client(config) {
+  SynchronousClient(const ClientConfig& config) : Client(config),
+                                                  interarrival_timer_() {
     num_threads_ =
       config.outstanding_rpcs_per_channel() * config.client_channels();
     responses_.resize(num_threads_);
+
```
Comment (ctiller) [inline test/cpp/qps/client_sync.cc]: Could this move into client.h (with, maybe, a ```c bool NextTimeout(timepoint*) ``` helper to get the next timeout (or not).

Comment (vjpai) [inline test/cpp/qps/client_sync.cc]: Sure, I can do that. Something similar is needed in async anyway (today's project)

**Comment by vjpai** [comment]
Closing this for now, will reopen when ready for review.

**Pull request closed without merging**